src/mustache.rs

```diff
@@ -37,12 +37,12 @@ impl Mustache {
         }
     }
 
-    // TODO: infallible function, no need to return Result
-    pub fn parse(str: &str) -> anyhow::Result<Mustache> {
+    pub fn parse(str: &str) -> Mustache {
         let result = parse_mustache(str).finish();
+
         match result {
-            Ok((_, mustache)) => Ok(mustache),
-            Err(_) => Ok(Mustache::from(vec![Segment::Literal(str.to_string())])),
+            Ok((_, mustache)) => mustache,
+            Err(_) => Mustache::from(vec![Segment::Literal(str.to_string())]),
         }
```

Comment:
```suggestion
        parse_mustache(str)
            .finish()
            .map(|(_, mustache)| mustache)
            .unwrap_or(Mustache::from(vec![Segment::Literal(str.to_string())]))
```

let me know if it makes sense. personally, looks better, but I'd like a second opinion